pkg/services/publicdashboards/publicdashboard.go

```diff
@@ -15,7 +15,7 @@ type Service interface {
 	BuildAnonymousUser(ctx context.Context, dashboard *models.Dashboard) (*models.SignedInUser, error)
 	GetPublicDashboard(ctx context.Context, accessToken string) (*models.Dashboard, error)
 	GetPublicDashboardConfig(ctx context.Context, orgId int64, dashboardUid string) (*PublicDashboard, error)
-	SavePublicDashboardConfig(ctx context.Context, dto *SavePublicDashboardConfigDTO) (*PublicDashboard, error)
+	SavePublicDashboardConfig(ctx context.Context, dto *SavePublicDashboardConfigDTO, dashboard *models.Dashboard) (*PublicDashboard, error)
```

Comment:
The service should be responsible for retrieving the dashboard instead of passing it in.